extension_templates/transformer.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+Extension template for transformers.
+
+Purpose of this implementation template:
+    quick implementation of new estimators following the template
+    NOT a concrete class to import! This is NOT a base class or concrete class!
+    This is to be used as a "fill-in" coding template.
+
+How to use this implementation template to implement a new estimator:
+- make a copy of the template in a suitable location, give it a descriptive name.
+- work through all the "todo" comments below
+- fill in code for mandatory methods, and optionally for optional methods
+- you can add more private methods, but do not override BaseEstimator's private methods
+    an easy way to be safe is to prefix your methods with "_custom"
+- change docstrings for functions and the file
+- ensure interface compatibility by testing forecasting/tests/test_all_forecasters
+        and forecasting/tests/test_sktime_forecasters
+- once complete: use as a local library, or contribute to sktime via PR
+
+Mandatory implements:
+    fitting         - _fit(self, X, y=None)
+    transformation  - _transform(self, X, y=None)
+
+Optional implements:
+    inverse transformation      - _inverse_transform(self, X, y=None)
+    fitted parameter inspection - get_fitted_params()
+
+State:
+    fitted model/strategy   - by convention, any attributes ending in "_"
+    fitted state flag       - is_fitted (property)
+    fitted state inspection - check_is_fitted()
+
+Testing:
+    get default parameters for test instance(s) - get_test_params()
+    create a test instance of estimator class   - create_test_instance()
+"""
+
+# todo: uncomment the following line, enter authors' GitHub IDs
+# __author__ = [authorGitHubID, anotherAuthorGitHubID]
+
+from sktime.transformations.base import BaseTransformer
+
+# todo: add any necessary imports here
+
+
+class MyTransformer(BaseTransformer):
+    """Custom transformer. todo: write docstring.
+
+    todo: describe your custom transformer here
+
+    Hyper-parameters
+    ----------------
+    parama : int
+        descriptive explanation of parama
+    paramb : string, optional (default='default')
+        descriptive explanation of paramb
+    paramc : boolean, optional (default= whether paramb is not the default)
+        descriptive explanation of paramc
+    and so on
+
+    Components
+    ----------
+    est : sktime.estimator, BaseEstimator descendant
+        descriptive explanation of est
+    est2: another estimator
+        descriptive explanation of est2
+    and so on
+    """
+
+    # todo: fill out estimator tags here
+    #  tags are inherited from parent class if they are not set
+    # todo: define the transformer scitype by setting the tags
+    #  the "transformer scitype" is determined by the tags
```

Comment:
I don't see the redundancy - you are removing multiple lines with complementary information?

Reply:
This is very minor. Adjust as you see fit.

* I think Line 1 "todo: fill out estimator tags here" is redundant as the instruction "define scitype by setting tags" already tells an extender they need to fill out the tags.

* I think Line 2 "tags are inherited from parent class if they are not set" is redundant, but only because you have copied all the tags into the template, i.e. all the tags are set.

* I think Line 4 `the "transformer scitype" is determined by the tags` is redundant because the instruction says that setting tags defines the scitype.

Reply:
updated the todos, hope it's clearer now.

Reply:
Yup, looks good